index.html

```diff
@@ -2326,6 +2327,102 @@ <h2>Definition of Roles</h2>
 				</tbody>
 			</table>
 		</div>
+		<div class="role" id="comment">
+			<rdef>comment</rdef>
+			<div class="role-description">
+				<p>A comment contains a remark expressing a reaction to other content within the page.</p>
+				<p>The relationship of comments to other content SHOULD be exposed as follows:</p>
+				<ol>
+					<li>If the comment refers to other content within the page, the source content SHOULD use <pref>aria-details</pref> to point to the related comment.</li>
+					<li>If the comment is a reply to another comment:
+						<ul>
+							<li>If all ancestor comments are available in the DOM, the reply comment SHOULD be a semantic descendant of the comment it relates to, either as a DOM descendant element, or via <pref>aria-owns</pref>. In this case, user agents MUST automatically compute the level and expose it to assistive technologies. This is the preferred method for defining
+								the comment hierarchy, as it creates a single root comment that can be referred to by commented content via <pref>aria-details</pref>.</li>
+							<li>Alternatively, if all ancestor comments are not in the DOM, such as when comments are paginated, the hierarchical
+								level MAY be indicated via <pref>aria-level</pref>.</li>
+						</ul>
+					</li>
+				</ol>
```

Comment:
Add a paragraph with user agent requirements about aria-level, aria-posinset, and aria-setsize here. I suppose this is not the place to specify how the computation should be done is it? I am wondering about the situations where the author has specified it on some but not others? What should the user agent do? Is it sufficient for the author to declare aria-level on the first comment element in the DOM? would the user agent then be able to compute subsequent values based on that value? BTW, we do not have normative user agent MUST statements for aria-level, aria-posinset, or aria-setsize for any other element that supports them. I think we should. but, we do not.

```suggestion
				<p>If the author has not explicitly declared <pref>aria-level</pref>, <pref>aria-posinset</pref>, or <pref>aria-setsize</pref> for a <code>comment</code> element, user agents MUST automatically compute the values and expose them to assistive technologies.</p>
```

Reply:
Hi all, I posted a new version. It is inspired by @mcking65's version but still fairly different in a few place. I think I addressed concerns but improved readability. I also made sure the container case wasn't related to what the commented content was. Please take a look.